File: typescript/brands.ts
export default interface Brand{
    setBrand(): string;
    boot():void;
}

export class Dell implements Brand {
    setBrand(): string {
        return "Keyboard input";
    }
    boot(){
        console.log("Dell  booting...");
    }

}

export class IBM implements Brand {
    setBrand(): string {
        return "Brand: IBM";
    }
    boot(){
        console.log("IBM  booting...");
    }
}

export class Lenovo implements Brand {
    setBrand(): string {
        return  "Brand: Lenovo";
    }

    boot(){
        console.log("Lenovo  booting...");
    }
}

export class Hp implements Brand {
    setBrand(): string {
        return "Brand: Hp";
    }
    boot(){
        console.log("HP  booting...");
    }
}

File: python/brands.py
from typing import Protocol


class Brand(Protocol):
    def boot(self):
        ...

    def set_brand(self) -> str: ...


class Dell(Brand):
    def set_brand(self) -> str:
        return "Brand: Dell"
    
    def boot(self):
        print("Dell booting...")


class IBM(Brand):
    def set_brand(self) -> str:
        return "Brand: IBM"
    
    def boot(self):
        print("IBM booting...")


class HP(Brand):
    def set_brand(self) -> str:
        return "Brand: HP"
    
    def boot(self):
        print("HP booting...")


class Lenovo(Brand):
    def set_brand(self) -> str:
        return "Brand: Lenovo"
    
    def boot(self):
        print("Lenovo booting...")
